conf/kubernetes-shuffle-service.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  labels:
+    app: spark-shuffle-service
+    version: 2.1.0
+  name: shuffle
+spec:
+  template:
+    metadata:
+      labels:
+        app: spark-shuffle-service
+    spec:
+      volumes:
+        - name: temp-volume
+          hostPath:
+            path: '/tmp'
```

Comment:
Trying to understand your points. 

> it doesn't deploy an entire application
Helm can be used for deploying a service. I consider the spark shuffle service a service independent of individual spark jobs.

Or, by "an entire application", do you mean the fact that the docker image may not be what the user wants?

> the whole overhead of helm/tiller would be prohibitive

By "overhead", do you mean the "setup steps" such as downloading helm and doing `$ helm init`? Or do you mean the "runtime" cpu, etc overhead?

If you mean "setup steps", it may not be a lot compared with the alternative of modifying this yaml file on her own and doing `$ kubectl create`. At least that was my experience. Another thing to consider is mucking with yaml can easily lead to syntax errors, and troubleshooting hours. Helm chart is a black box shielding inexperienced users from that.

Anyway, I wouldn't block this PR for not having helm. I just think it is nice way to go for the future iterations of this.

Independent of helm or not, I still think putting `/tmp` here is a not a good example.

One more related point is that we may have multiple dirs in this as an example. Shuffle is seek intensive. Using multiple disks is quite common. We may want to show how it can be done as an example.